### ISSUE TITLE
incompatible with ember-computed-decorators

### ISSUE THREAD

**Issue by bartocc** [ember-cli/eslint-plugin-ember#42]
When using https://github.com/rwjblue/ember-computed-decorators, you'll get lint errors because multiline computed properties are now seen as methods and according to the order-in-controllers rule for example, should go below actions.

**Comment by michalsnik** [comment]
Hey, It's not a usual scenario so I don't think we'll work on this issue anytime soon. But if you want to help - feel free. Any PRs are very welcome :)

**Comment by bartocc** [comment]
I'll try to dedicate some time to put up a PR.

Is there any slack channel dedicated to eslint-plugin-ember in case I need
some help ?

2017-03-24 16:14 GMT+01:00 Michał Sajnóg <notifications@github.com>:

> Hey, It's not a usual scenario so I don't think we'll work on this issue
> anytime soon. But if you want to help - feel free. Any PRs are very welcome
> :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <http://clicks.crx.io/wf/click?upn=S8NtJlhzKsbL5BEVJrBaBUgz1uHV9901UEOCJ2xUGQIVhY62j-2FWu1Zra-2Bz0aCfFrRb9cWmEfOr8l-2FwSsJWF4WPIb4-2BstVM3Y9vELg-2Flnze1uNeU-2Fa-2BdqZ-2Ftdy7ysvjq9_s5UH8bhnBaopYjthcNSRcS0tHwGDbWoSl-2FKHoPS6OPPIWSVKUh9-2FP9jqMJ9Z3Er5t4gZAZ2V-2Bg0ecOmhy4rdB3kdKDfqQaR6lY9BqCGrV8ubV85isKoE1ET35j7qSD5y-2Fyl9RDgR835njytAb32gyrNQWcWEw3iB3xqC3MCNTGR5MbqhuJLZmuCIsU-2FoRTbswYjkCcII7DsAAqHDZRt2JJLrTJaSW3SO8FCzcEfHNbbQnAspRl3KyL-2Fv7FEjjgW7qeD-2BKjt8KSqOjlz9gShheMdi8MyjIYPkpIE2CcF-2B5DNHTEdnlFOAR0IB9Pwyp-2FZqcRHqB1LjuekvDdciZ5cQsA-3D-3D>,
> or mute the thread
> <http://clicks.crx.io/wf/click?upn=S8NtJlhzKsbL5BEVJrBaBSSbuIiZAgCCd1YT5JjUGl-2BNz90vdAsqQUu80bVdTYECLFiUqhPP9hMJNg9-2FFJj-2BV3iufD00a-2F5PRAep6kACwzauWdjhDdjnn5LRuE01r2vCnMDjO9INgD3bGpYty-2BdGwQ-3D-3D_s5UH8bhnBaopYjthcNSRcS0tHwGDbWoSl-2FKHoPS6OPPIWSVKUh9-2FP9jqMJ9Z3Er5t4gZAZ2V-2Bg0ecOmhy4rdB3kdKDfqQaR6lY9BqCGrV8ubV85isKoE1ET35j7qSD5y-2Fyl9RDgR835njytAb32gyrNQWcWEw3iB3xqC3MCNTGR5MbqhuJLZmuCIsU-2FoRTbsuvf5ZU1sqNsc2fMyh1MhDoWCbnAchnbK58F7KoF2OqYLZOmed6G6RBVGFSBst-2Fh1lYbiZlQb-2B1a0TTFL6Z1ygkqafXBw6mbV6f4PHm-2B6gBHoyo4Vr0i6ejPCT-2B3bKAkYFQjJkT-2FpCDXhZ3M4zw2HPw-3D-3D>
> .
>



-- 

-
[image: Julien Palmas]
<http://clicks.crx.io/wf/click?upn=S8NtJlhzKsbL5BEVJrBaBe-2FGEnkz6ZvSgIEFjbetRNnMXXsDx5s-2FniDMCo7Ru-2B9RItprvbmbGozkFV8muwzQ5QhB-2Fg7DHLQeEgGwXP4U76w-3D_s5UH8bhnBaopYjthcNSRcS0tHwGDbWoSl-2FKHoPS6OPPIWSVKUh9-2FP9jqMJ9Z3Er5t4gZAZ2V-2Bg0ecOmhy4rdB3kdKDfqQaR6lY9BqCGrV8ubV85isKoE1ET35j7qSD5y-2Fyl9RDgR835njytAb32gyrNQWcWEw3iB3xqC3MCNTGR5MbqhuJLZmuCIsU-2FoRTbss2-2FFD-2FEk3Q5qIt3-2BT2Fhytlrm5z9WpDn-2BLae56gw5h5RwjSk9cl3r83gv-2BYYOW6bKAVXX8XF89Y7cCTKw33PsU1GC-2Fc-2Bhm7aM1jTWJe9jMeMVCG3FEZ-2FmU-2FF0l205d-2BMHQQER7zC3rPb86UvcLYJBg-3D-3D>

**Comment by michalsnik** [comment]
Finally there is a new channel in https://embercommunity.slack.com/ called #dev-linting. You can reach me there in case of any doubts :)

**Comment by alexdiliberto** [comment]
Looked into this a bit this past weekend, it's a more involved fix that includes updating the netguru eslint rules to handle potential `babel-eslint` parser usage (because the AST's are different). Anyone with the experience or time to make these changes would be a big help /cc @bartocc.

**Comment by Kerrick** [comment]
Another incompatibility with ember-computed-decorators (now ember-decorators) is that `ember/avoid-leaking-state-in-components` fails the "[Real-world get/set syntax](https://ember-decorators.github.io/ember-decorators/docs/function/addon/object/index.js.html#-quot-real-world-get-set-syntax-quot-)" from the ED docs.

**Comment by michalsnik** [comment]
Ok I think it's not necessarily a problem with `ember-decorators` @Kerrick but rather ES6 Class syntax and native getters/setters. I'm aware of the future, and that one day we'll probably start writing classes in ember, but it's not a small pill to swallow.

While introducing support for classes we'll still need to support old way. Before we introduce this change we'd rather want to make sure that this will actually be something natively available in ember and that related RFC is in last comment period.

It's a good time to spike it and think about possible issues though.
I'm not sure I have enough capacity to handle this soon, but I'll add it my to-do list just for the record.

**Comment by michalsnik** [comment]
Closing in favor of #96

**Comment by Kerrick** [comment]
@michalsnik That's not entirely true -- Ember Decorators also supports [using get/set syntax on non-ES6 classes](https://github.com/ember-decorators/ember-decorators/tree/v0.3.0#real-world-getset-syntax), like this:

```js
import Ember from 'ember';
import computed from 'ember-computed-decorators';

export default Ember.Component.extend({
  @computed('foo', 'bar')
  fooBar: {
    get(foo, bar) {
      return `${foo} ${bar}`;
      },
    set(value) {
      const [foo, bar] = value.split(' ');
      this.setProperties({ foo, bar });
      return value;
    }
  }
});
```

This will still fail `ember/avoid-leaking-state-in-components`, but won't be solved by #96's focus on native ES6 classes.
